pkg/controller/mpi_job_controller.go

```diff
@@ -503,24 +505,13 @@ func (c *MPIJobController) syncHandler(key string) error {
 	// cleanup and stop retrying the MPIJob.
 	if isFinished(mpiJob.Status) && mpiJob.Status.CompletionTime != nil {
 		if isCleanUpPods(mpiJob.Spec.RunPolicy.CleanPodPolicy) {
-			// set worker StatefulSet Replicas to 0.
-			if err := c.deleteWorkerPods(mpiJob); err != nil {
-				return err
-			}
-			initializeMPIJobStatuses(mpiJob, kubeflow.MPIReplicaTypeWorker)
-			if c.gangSchedulerName != "" {
-				if err := c.deletePodGroups(mpiJob); err != nil {
-					return err
-				}
-			}
-			mpiJob.Status.ReplicaStatuses[common.ReplicaType(kubeflow.MPIReplicaTypeWorker)].Active = 0
-			return c.updateStatusHandler(mpiJob)
+			return cleanUpPods(mpiJob, c)
 		}
 		return nil
 	}
 
 	// first set StartTime.
-	if mpiJob.Status.StartTime == nil {
+	if mpiJob.Status.StartTime == nil && !isMPIJobSuspended(mpiJob) {
```

Comment:
if the job was suspended, we should reset the StartTime. Double check how we do it in the job controller.

Reply:
Done - I've added the JobSuspended condition and replicated the semantics around StartTime. 

However, the `.spec.runPolicy.activeDeadlineSeconds` is actually respected for MPIJobs via batch.Job: https://github.com/kubeflow/mpi-operator/blob/382da780a7a9a23e6c9fd9dc8c93ba77aba81c48/pkg/controller/mpi_job_controller.go#L1320. 

This means that the changes aren't strictly required to enforce the timeout. WDYT?